document/src/vespa/document/fieldvalue/fieldvalue.h

```diff
@@ -84,6 +84,13 @@ class FieldValue : public vespalib::Identifiable
     virtual int compare(const FieldValue& other) const;
 
     /**
+     * Same as normal compar, but this one expects the types to be equal
+     * @param other
+     * @return See compare
+     */
+    virtual int fastCompare(const FieldValue& other) const;
```

Comment:
I'm assuming that `fastCompare` is added at least partially due to the overhead of `operator==` (which itself is virtual) on `DataType`? Could we also optimize the common case of identical types from the same repo generally here by refactoring `DataType` from
```c++
virtual bool operator==(const DataType&) const;
```
to
```c++
virtual int compare(const DataType&) const:

bool operator==(const DataType& rhs) const {
    if (this != &rhs) {
        return (compare(rhs) == 0);
    }
    return true;
}
```
Not saying we should do this now, just airing the idea.

Could possible also short-circuit `FieldValue::compare` into `fastCompare` if type equivalence holds.